app/controllers/application_controller/advanced_search.rb

```diff
@@ -48,7 +48,7 @@ def adv_search_button_saveid
       add_flash(_("Search Name is required"), :error) if params[:button] == 'saveit'
       false
     else
-      s = @edit[@expkey].build_search(@edit[:new_search_name], @edit[:search_type], session[:userid])
+      s = @edit[@expkey].build_search(@edit[:new_search_name], current_user.admin_user? || current_user.super_admin_user? ? @edit[:search_type] : nil, session[:userid])
```

Comment:
@Nadkine `admin_user?` method does not exist, you should only check for `current_user.super_admin_user?`